hypothesis-python/src/hypothesis/database.py

```diff
@@ -22,12 +22,30 @@
 import warnings
 from hashlib import sha1
 
-from hypothesis.configuration import storage_directory
+from hypothesis.configuration import hypothesis_home_dir
```

Comment:
Having grepped through uses of `storage_directory()`, it doesn't look like there are *any* cases where we want the directory created there...  and it's the only use of `hypothesis_home_dir()`.  Let's inline that logic and delete the single-use `tmpdir()` helper for simplicity.

Then all our calls can just be `storage_directory("some", "paths")` and we don't need to wrap it in `os.path.join()`.